src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -300,74 +298,25 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
+    void ExtensionPalette::_closeChat(const Windows::Foundation::IInspectable& /*sender*/,
+                                      const Windows::UI::Xaml::RoutedEventArgs& /*args*/)
     {
-        if (Visibility() != Visibility::Collapsed)
-        {
-            _close();
-        }
+        Visibility(Visibility::Collapsed);
     }
 
     void ExtensionPalette::_backdropPointerPressed(const Windows::Foundation::IInspectable& /*sender*/,
                                                    const Windows::UI::Xaml::Input::PointerRoutedEventArgs& e)
     {
+        _setFocusAndPlaceholderTextHelper();
         e.Handled(true);
     }
 
-    // Method Description:
-    // - The purpose of this event handler is to hide the palette if it loses focus.
-    // We say we lost focus if our root element and all its descendants lost focus.
-    // This handler is invoked when our root element or some descendant loses focus.
-    // At this point we need to learn if the newly focused element belongs to this palette.
-    // To achieve this:
-    // - We start with the newly focused element and traverse its visual ancestors up to the Xaml root.
-    // - If one of the ancestors is this ExtensionPalette, then by our definition the focus is not lost
-    // - If we reach the Xaml root without meeting this ExtensionPalette,
-    // then the focus is not contained in it anymore and it should be dismissed
-    // Arguments:
-    // - <unused>
-    // Return Value:
-    // - <none>
-    void ExtensionPalette::_lostFocusHandler(const Windows::Foundation::IInspectable& /*sender*/,
-                                             const Windows::UI::Xaml::RoutedEventArgs& /*args*/)
+    void ExtensionPalette::_queryBoxGotFocusHandler(const Windows::Foundation::IInspectable& /*sender*/,
+                                                    const Windows::UI::Xaml::RoutedEventArgs& /*args*/)
     {
-        const auto flyout = _queryBox().ContextFlyout();
-        if (flyout && flyout.IsOpen())
-        {
-            return;
-        }
-
-        auto root = this->XamlRoot();
-        if (!root)
-        {
-            return;
-        }
-
-        auto focusedElementOrAncestor = Input::FocusManager::GetFocusedElement(root).try_as<DependencyObject>();
-        while (focusedElementOrAncestor)
-        {
-            if (focusedElementOrAncestor == *this)
-            {
-                // This palette is the focused element or an ancestor of the focused element. No need to dismiss.
-                return;
-            }
-
-            // Go up to the next ancestor
-            focusedElementOrAncestor = winrt::Windows::UI::Xaml::Media::VisualTreeHelper::GetParent(focusedElementOrAncestor);
-        }
-
-        // We got to the root (the element with no parent) and didn't meet this palette on the path.
-        // It means that it lost the focus and needs to be dismissed.
-        _close();
+        _ActiveControlInfoRequestedHandlers(nullptr, nullptr);
```

Comment:
this design is not ideal - it effectively *assumes* that the recipient of this event will populate `ActiveCommandline`, via some "behind the scenes magic".

I'm not asking that you change it, but more FYI. "I send out a random event, and then check my own data" is a VEEEERY shaky implicit contract.